lib/ui/volatile_path_tracker.cc

```diff
@@ -15,71 +15,58 @@ void VolatilePathTracker::Insert(std::shared_ptr<TrackedPath> path) {
   FML_DCHECK(ui_task_runner_->RunsTasksOnCurrentThread());
   FML_DCHECK(path);
   FML_DCHECK(path->path.isVolatile());
+  FML_DCHECK(!path->erased);
   if (!enabled_) {
     path->path.setIsVolatile(false);
     return;
   }
-  paths_.insert(path);
+  paths_.push_back(path);
 }
 
 void VolatilePathTracker::Erase(std::shared_ptr<TrackedPath> path) {
   if (!enabled_) {
     return;
   }
   FML_DCHECK(path);
-  if (ui_task_runner_->RunsTasksOnCurrentThread()) {
-    paths_.erase(path);
-    return;
-  }
-
-  std::scoped_lock lock(paths_to_remove_mutex_);
-  needs_drain_ = true;
-  paths_to_remove_.push_back(path);
+  std::scoped_lock lock(paths_mutex_);
```

Comment:
I could potentially check whether we're on the UI task runner here to avoid grabbing the lock, but:

- IME this often gets called from a non-UI task runner during parallel GC activity
- It's not clear how much faster the branch to check runs on current thread would really be vs just getting the lock.

Reply:
It's in situations like this that I really wish we had the static thread safety analysis decorated sync utilities of our own.